File: 2871-double-a-number-represented-as-a-linked-list/double-a-number-represented-as-a-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def doubleIt(self, head):
        if head.val > 4:
            head = ListNode(0, head)
        
        tmp = head
        while tmp is not None:
            tmp.val = (tmp.val * 2) % 10
            if tmp.next is not None and tmp.next.val > 4:
                tmp.val += 1
            
            tmp = tmp.next
        
        return head

File: 2871-double-a-number-represented-as-a-linked-list/double-a-number-represented-as-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode doubleIt(ListNode head) {
        if (head.val > 4) {
            head = new ListNode(0, head);
        }
        
        ListNode tmp = head;
        while (tmp != null) {
            tmp.val = (tmp.val * 2) % 10;
            if (tmp.next != null && tmp.next.val > 4) {
                tmp.val++;
            }
            
            tmp = tmp.next;
        }
        
        return head;
    }
}